sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java

```diff
@@ -1820,10 +1827,138 @@ void testDetermineCharsetFromMessageFromHttpHeader(@TempDir final File dir) thro
+                    CommunicationLog.Direction.INBOUND,
+                    CommunicationLog.MessageType.REQUEST,
+                    communicationContext,
+                    messageStorage);
+            message.close();
+
+            // when
+            final Charset actualCharset = messageStorage.determineCharsetFromMessage(message);
+            assertEquals(1, messageStorage.getMessageEncodingErrorCount());
+
+            // then
+            assertEquals(Charset.forName("ISO-8859-13"), actualCharset);
+            verify(this.testRunObserver).invalidateTestRunWithoutReason();
+            verifyNoMoreInteractions(this.testRunObserver);
+        }
+    }
+
+    /**
+     * Ensures that determineCharsetFromMessage() is not called when enableEncodingCheck=false.
+     * @param dir - a temporal directory.
+     * @throws IOException - when something goes wrong.
+     */
     @Test
-    void testDetermineCharsetFromMessageFromHttpHeaderWithQuotes(@TempDir final File dir) throws IOException {
-        try (final MessageStorage messageStorage =
-                new MessageStorage(1, mock(MessageFactory.class), new HibernateConfigImpl(dir), this.testRunObserver)) {
+    public void testDetermineCharsetFromMessageFromHttpHeaderEncodingCheckDisabled(@TempDir final File dir)
+            throws IOException {
+        try (final MessageStorage messageStorage = new MessageStorage(
+                1, false, false, mock(MessageFactory.class), new HibernateConfigImpl(dir), this.testRunObserver)) {
+            // given
+            assertEquals(0, messageStorage.getMessageEncodingErrorCount());
+
+            final ListMultimap<String, String> headers = ArrayListMultimap.create();
+            headers.put("Content-Type", "application/xml; charset=ISO-8859-13");
+
+            final HttpApplicationInfo applicationInfo = new HttpApplicationInfo(headers, "transactionId", "requestURI");
+            final TransportInfo transportInfo =
+                    new TransportInfo("http", "localhost", 1234, "remotehost", 4567, List.of());
+            final CommunicationContext communicationContext = new CommunicationContext(applicationInfo, transportInfo);
+            final Message message = new Message(
+                    CommunicationLog.Direction.INBOUND,
+                    CommunicationLog.MessageType.REQUEST,
+                    communicationContext,
+                    messageStorage);
+            final OutputStreamWriter writer = new OutputStreamWriter(message, StandardCharsets.UTF_8);
+            writer.write("some body");
+            writer.flush();
+            writer.close();
+            message.close();
+
+            // when
+            messageStorage.flush();
+            assertEquals(0, messageStorage.getMessageEncodingErrorCount());
+
+            // then
+            verifyNoInteractions(this.testRunObserver);
+        }
+    }
+
+    /**
+     * Ensures that the test run is not invalidated when an MIME type error is detected
```

Comment:
The documentation no longer matches. Test run is invalidated when a MIME type error is detected.

Reply:
Not any more. Now the description matches again.